satpy/readers/yaml_reader.py

```diff
@@ -823,19 +835,31 @@ def _load_dataset_with_area(self, dsid, coords, **kwargs):
         if not file_handlers:
             return
 
-        area = self._load_dataset_area(dsid, file_handlers, coords, **kwargs)
-
         try:
             ds = self._load_dataset_data(file_handlers, dsid, **kwargs)
         except (KeyError, ValueError) as err:
             logger.exception("Could not load dataset '%s': %s", dsid, str(err))
             return None
 
+        coords = self._assign_builtin_coords(coords, ds)
+
+        area = self._load_dataset_area(dsid, file_handlers, coords, **kwargs)
+
         if area is not None:
             ds.attrs['area'] = area
             ds = add_crs_xy_coords(ds, area)
         return ds
 
+    @staticmethod
+    def _assign_builtin_coords(coords, ds):
```

Comment:
Thoughts on changing the name to something like `_assign_existing_coords` or `_assign_coords_from_data_array`? To me `builtin` is ambiguous.